### DISCUSSION TITLE
How do I listen for sub-page requests？

### DISCUSSION THREAD

**Discussion by hanfeige** [MicrosoftEdge/WebView2Feedback#3992] [Q&A]
When a page has an IFRame (and possibly a cross-domain) and cannot get a network request, I try two methods

1, use WebResourceResponseReceived
```
        await webView.EnsureCoreWebView2Async(null);
        webView.CoreWebView2.WebResourceResponseReceived += Test_WebResourceResponseReceived;
```
2. Use DevToolsProtocol
```
        await webView.EnsureCoreWebView2Async(null);
        webView.CoreWebView2.Settings.AreDevToolsEnabled = true;
        await webView.CoreWebView2.CallDevToolsProtocolMethodAsync("Network.enable", "{}");
        var eventReceiver = webView.CoreWebView2.GetDevToolsProtocolEventReceiver("Network.responseReceived");
        eventReceiver.DevToolsProtocolEventReceived += Test_ResponseReceived;
```

I try to use the Playwright Edge way, through the monitoring page, get all requests, including the IFrame, so speculation has nothing to do with the Edge should be, but I need to use the WPF, Playwright doesn't suit me

But I can only get home page requests. What do I do？

**Comment by vbryh-msft** [comment]
@hanfeige cross-domain iframes support for WebResourceRequested is available in prerelease package - you would need to use [AddWebResourceRequestedFilter](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.addwebresourcerequestedfilter?view=webview2-dotnet-1.0.1369-prerelease#microsoft-web-webview2-core-corewebview2-addwebresourcerequestedfilter(system-string-microsoft-web-webview2-core-corewebview2webresourcecontext-microsoft-web-webview2-core-corewebview2webresourcerequestsourcekinds)) and pass third parameter as Document.
